src/tasks/Readme.md

```diff
@@ -0,0 +1,58 @@
+# Custom Azure DevOps tasks
+
+## Prerequisites
+
+- [Node.js](http://nodejs.org) 4.0.x or later and NPM (included with the installer)
+
+## Creating a custom task
+
+### Creating a new custom task
+
+1. Task folder layout in this repo follows this format...
+
+    ```text
+    src/tasks/tasks
+      /[TaskName]V[TaskVersion]
+    ```
+
+2. General task creation instructions are located [here](https://docs.microsoft.com/en-us/azure/devops/extend/develop/add-build-task?view=vsts#step-1-create-the-custom-task)
+
+3. After writing your task, add an entry for your task into `src/tasks/make-options.json` in the "tasks" element.
+
+4. Build the tasks using either `/build-tasks.cmd` or `src/tasks/build.cmd`
+
+### Packaging the task in the extension
+
+Extension packaging is controlled via the `vss-extension.json` file.
+
+[Extension layout guidelines](https://docs.microsoft.com/en-us/azure/devops/extend/develop/integrate-build-task?view=vsts#multiple-version-layout)
+
+1. Modify `vss-extension.json` and add appropriate entries for your created task.
+
+2. Currently, you must manually increment the version number for the manifest you are creating
+
+```json
+    "manifestVersion": 1,
+    "id": "dnceng-build-release-tasks",
+    "name": "Dnceng Build Tasks",
+    "version": "0.0.1", <== manually rev to "0.0.2" before packaging
+    "publisher": "dotnet-dnceng",
+```
+
+### Publishing
+
+Note: See [Developer publishing flow](#developer-publishing-flow) before publishing so that you can validate your changes before exposing them to dnceng.
+
+We have a [dotnet-dnceng publisher](https://marketplace.visualstudio.com/manage/publishers/dotnet-dnceng) defined in the VS Marketplace.
+
+1. Find the "Dnceng Build Tasks" extension, select "..." and choose "Update"
+
+2. Upload the extension you created from the `.artifacts/extensions` folder
+
+3. The extension will now get published to all participating Azure DevOps accounts
+
+### Developer publishing flow
+
+The "dotnet-dnceng" publisher mentioned in [Publishing](#publishing) will immediately update the dnceng account with any extension changes you have made.
+
+For testing purposes, you should create your own publisher which publishes to your own organization and use that flow to validate your task changes before using the dotnet-dnceng publisher to expose your changes to the dnceng organization.
```

Comment:
Should we link to a doc on how to do this?

Reply:
hmmm... I guess we could expand on this, but what's in this short paragraph is generally the process.